drivers/rotator/nframe.cpp

```diff
@@ -0,0 +1,709 @@
+/*
+  nFrameRotator Focuser
+
+  Copyright(c) 2019 Jasem Mutlaq. All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "nframe.h"
+
+#include "indicom.h"
+
+#include <cstring>
+#include <termios.h>
+#include <memory>
+#include <thread>
+#include <chrono>
+
+static std::unique_ptr<nFrameRotator> rotator(new nFrameRotator());
+
+
+nFrameRotator::nFrameRotator()
+{
+    setVersion(1, 2);
+    RI::SetCapability(ROTATOR_CAN_ABORT |
+                      ROTATOR_CAN_SYNC
+                     );
+}
+
+bool nFrameRotator::initProperties()
+{
+    INDI::Rotator::initProperties();
+
+
+
+    // Stepping Modes
+    IUFillSwitch(&SteppingModeS[STEPPING_WAVE], "STEPPING_WAVE", "Wave", ISS_OFF);
+    IUFillSwitch(&SteppingModeS[STEPPING_HALF], "STEPPING_HALF", "Half", ISS_OFF);
+    IUFillSwitch(&SteppingModeS[STEPPING_FULL], "STEPPING_FULL", "Full", ISS_ON);
+    IUFillSwitchVector(&SteppingModeSP, SteppingModeS, 3, getDeviceName(), "STEPPING_MODE", "Mode",
+                       STEPPING_TAB, IP_RW, ISR_1OFMANY, 0, IPS_OK);
+
+    // Stepping Phase
+    IUFillNumber(&SteppingPhaseN[0], "PHASES", "Wiring", "%.f", 0, 2, 1, 0);
+    IUFillNumberVector(&SteppingPhaseNP, SteppingPhaseN, 1, getDeviceName(), "STEPPING_PHASE", "Phase",
+                       STEPPING_TAB, IP_RW, 0, IPS_OK);
+
+
+    IUFillNumber(&FocusSpeedN[0], "FOCUS_SPEED_VALUE", "Step Rate", "%3.0f", 0.0, 255.0, 1.0, 255.0);
+    //    IUFillNumberVector(&FocusSpeedNP, FocusSpeedN, 1, getDeviceName(), "FOCUS_SPEED", "Speed", MAIN_CONTROL_TAB, IP_RW, 60, IPS_OK);
+    IUFillNumberVector(&FocusSpeedNP, FocusSpeedN, 1, getDeviceName(), "FOCUS_SPEED", "Speed", MAIN_CONTROL_TAB, IP_RW, 0,
+                       IPS_OK);
```

Comment:
What for?

Reply:
These are the current stepper rate, just copied over from the nstep focus driver, will max into Rotator nomenclature.